### PR TITLE
fix(scripts): add dry run release and fix major bump for python

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: -

### Changes included:

add dry run option for the release which allows testing the lts in a better manner
also fix pre-release bump for python versions, which is not semver compliant

### CODE REVIEW

File: scripts/cli/index.ts [hunk 1]
```diff
@@ -212,8 +200,9 @@ program
   .description('Releases the client')
   .addArgument(args.languages)
   .option(flags.verbose.flag, flags.verbose.description)
-  .option(flags.major.flag, flags.major.description)
-  .action(async (langArgs: LangArg[], { verbose, major }) => {
+  .option('-m, --major', 'triggers a major release for the given language list')
+  .option('-d, --dryRun', 'does not push anything to GitHub')
```
Comment (millotp) [inline scripts/cli/index.ts:204]:
```suggestion
  .option('-d, --dry-run', 'does not push anything to GitHub')
```
?

File: scripts/release/createReleasePR.ts [hunk 2]
```diff
@@ -44,6 +44,8 @@ dotenv.config({ path: ROOT_ENV_PATH });
 
 export const COMMON_SCOPES = ['specs', 'clients'];
 
+const preReleaseRegExp = new RegExp(/\d\.\d\.\d(\.?a(lpha\.)?\d+|\.?b(eta\.)?\d+)$/);
```
Comment (millotp) [inline scripts/release/createReleasePR.ts:47]: :face_vomiting:

File: scripts/release/createReleasePR.ts [hunk 3]
```diff
@@ -184,11 +186,13 @@ export function getNextVersion(current: string, releaseType: semver.ReleaseType
   // python pre-releases have a pattern like `X.Y.ZaN` for alpha or `X.Y.ZbN` for beta
```
Comment (millotp) [inline scripts/release/createReleasePR.ts:186]: maybe the comment should follow the regex

**Review by millotp — COMMENTED** [review]
:kermit_typing:

File: scripts/release/createReleasePR.ts (hunk 2 above)
Comment (shortcuts) [reply scripts/release/createReleasePR.ts:47]: ahahah yes this one is pretty ugly I admit

File: scripts/release/createReleasePR.ts (hunk 3 above)
Comment (shortcuts) [reply scripts/release/createReleasePR.ts:186]: true

File: scripts/cli/index.ts (hunk 1 above)
Comment (shortcuts) [reply scripts/cli/index.ts]: ah yes true

**Review by millotp — APPROVED** [review]
gg !

**Pull request merged**